index.js

```diff
@@ -0,0 +1,220 @@
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (item) => {
+          const card = renderCard(item);
+          cardSection.addItem(card);
+        },
+      },
+      selectors.cardSection
+    );
+    cardSection.renderItems();
+    userInfo.setUserInfo(user);
+    userInfo.setUserAvatar(user.avatar);
+  })
+  .catch((err) => {
+    console.error(`Error: ${err}`);
+  });
+
+// User Profile
+const updateProfilePopup = new PopupWithForm(
+  selectors.profilePopup,
+  handleProfileSubmit
+);
+updateProfilePopup.setEventListeners();
+
+// New Card
+const newCardPopup = new PopupWithForm(
+  selectors.cardPopup,
+  handleAddCardFormSubmit
+);
+newCardPopup.setEventListeners();
+
+// Render Card
+function renderCard(item) {
+  const card = new Card(
+    item,
+    selectors.cardTemplate,
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeIcon
+  );
+  return card.generateCard();
+}
+
+// Delete Card
+const deletePopup = new PopupConfirm(
+  selectors.deletePopup,
+  handleDeleteCard
+);
+deletePopup.setEventListeners();
+
+// Edit Avatar
+const avatarPopup = new PopupWithForm(
+  selectors.avatarPopup,
+  handleEditAvatar
+);
+avatarPopup.setEventListeners();
+
+// Event Listeners
+addCardAddButton.addEventListener("click", () => {
+  newCardPopup.open();
+  cardFormValidator.toggleButtonState();
+});
+
+updateProfileButton.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileHeadingInput.value = userData.name;
+  profileDescriptionInput.value = userData.about;
+  editProfileFormValidator.resetValidation();
+  updateProfilePopup.open();
+});
+
+updateAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// Functions
+function handleValidation(form) {
+  form.enableValidation();
+}
+
+function handleAddCardFormSubmit(cardData) {
+  newCardPopup.showButtonProgress(true);
+  api
+    .addNewCard(cardData.name, cardData.link)
+    .then((res) => {
+      const card = renderCard(res);
+      cardSection.addItem(card);
+      newCardPopup.reset();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      newCardPopup.showButtonProgress(false);
+    });
+}
+
+function handleImageClick(name, link) {
+  newImagePopup.open(name, link);
+}
+
+function handleProfileSubmit(userData) {
+  updateProfilePopup.showButtonProgress(true);
+  api
+    .updateUserInfo(userData.name, userData.about)
+    .then((user) => {
+      userInfo.setUserInfo(user);
+      updateProfilePopup.reset();
+      updateProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      updateProfilePopup.showButtonProgress(false);
+    });
+}
+
+function handleDeleteCard(cardData) {
+  deletePopup.open();
+  deletePopup.setSubmitAction(() => {
+    deletePopup.showButtonProgress(true);
+    api
+      .deleteCard(cardData._id)
```

Comment:
`cardData._id` will be `cardData.id`

Reply:
You haven't corrected my remark https://github.com/Marcus3278/se_project_aroundtheus-main/pull/6/files#r1615207582 


```js
      .deleteCard(cardData.id)

```